img_proof/ipa_cloud.py

```diff
@@ -349,7 +349,7 @@ def _run_test(self, test, ssh_config):
         if self.early_exit:
             options.append('-x')
 
-        args = '-v {} --ssh-config={} --hosts={} {}'.format(
+        args = '-v -s {} --ssh-config={} --hosts={} {}'.format(
```

Comment:
Did pytest's behavior change?

Reply:
We migrated to 15SP2 and with that we got version bumps of py and pluggy both of which are used by pytest in relation to the terminal writer handling. Also, the old mash server running on 15 had a version of pytest that was not from supported packages because we needed a version > what was available. Something very likely changed in the dependency tree of pytest that's causing the regression. I couldn't pinpoint exactly the changelog though. The issues related to pytest handling of stdout are well known and documented in multiple open bugs with have no solution.

Because img-proof is already redirecting stdout to a file there's no point for pytest to also do the redirect so at the very least this change doesn't hurt. I suspect it will solve or reduce the problem we are seeing.

Reply:
Now that I think of it more we probably still have a separate problem :grimacing: . Each instance of img-proof when it starts a pytest execution redirects stdout to that specific log file e.g. `/var/lib/mash/img_proof/results/azure/suse-sles-12-sp1-sapcal-v20200924/img-proof-lzbtx/20200924231134.log`. So we probably have an issue when multiple jobs are running that the stdout gets mixed up because these are running threaded and not separate processes. :facepalm:

Reply:
Sounds like another PR in the making

Reply:
Yep, will work on a new PR.